### PR TITLE
Refactor for Error Message class use

### PR DESCRIPTION
**Pull request by shas19**
Made following changes:

Moved all the methods which are related to creating error message to `ErrorBuilder` class. 
Used Error Message object as a parameter for `createErrorDescription`.
Some minor nits.

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/ErrorMessage.java [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+import java.util.Set;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Element;
+
 /** Contains error message string to be displayed and the message type from {@link MessageTypes}. */
 public class ErrorMessage {
+
+  static Config config;
+  static NullAway nullAway;
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/ErrorMessage.java]: No static fields please; they can lead to memory leaks and data races.  Also we shouldn't be keeping a pointer to the `NullAway` object here.  We should pass in just the state that we need.

Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/ErrorMessage.java]: Done 👍 
Separated the code into new `ErrorBuilder` class.

File: nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import java.util.Set;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Element;
+
+public class ErrorBuilder {
+
+  private final Config config;
+  private final String checkerName;
+  private final Set<String> allNames;
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java]: What is this field?  Pls document

File: nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Element;
+
+public class ErrorBuilder {
+
+  private final Config config;
+  private final String checkerName;
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java]: Should this be maybe `suppressionName`?  Looks like it's just used when generating a `@SuppressWarnings`.  Pls also add docs

File: nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessAnalysis.java [hunk 4]
```diff
@@ -48,8 +48,6 @@
 
   private final DataFlow dataFlow;
 
-  private static String OPTIONAL_PATH = "java.util.Optional";
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessAnalysis.java]: :+1:

**Comment by lazaroclapp** [comment]
Not a full review, but, eyeballing it, it looks fine to me, except the PR title/commit message has a typo: "Mesaage".

**Comment by shas19** [comment]
> Not a full review, but, eyeballing it, it looks fine to me, except the PR title/commit message has a typo: "Mesaage".

Oops! 😅

File: nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java (hunk 2 above)
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java]: 👍Done

File: nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java (hunk 3 above)
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java]: 👍Done

**Comment by lazaroclapp** [comment]
Hey @msridhar , do you have any unaddressed comments on this PR? I glanced at it and the refactor makes sense to me, given all tests pass, maybe worth merging it before #291 ? And releasing `0.7.0` with both?

**Comment by msridhar** [comment]
> Hey @msridhar , do you have any unaddressed comments on this PR? I glanced at it and the refactor makes sense to me, given all tests pass, maybe worth merging it before #291 ? And releasing `0.7.0` with both?

Sounds good

**Pull request merged**